src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -435,7 +435,33 @@ namespace winrt::TerminalApp::implementation
     void CommandPalette::_rootPointerPressed(Windows::Foundation::IInspectable const& /*sender*/,
                                              Windows::UI::Xaml::Input::PointerRoutedEventArgs const& /*e*/)
     {
-        _dismissPalette();
+        if (Visibility() != Visibility::Collapsed)
+        {
+            _dismissPalette();
+        }
+    }
+
+    // Method Description:
+    // - The purpose of this event handler is to hide the palette if it loses focus.
+    // Since the focus might move within the palette we need to register this handler for all focusable children
+    // Right now this is achieved by explicitly registering on the root on the highlighted text controls.
+    // Few notes:
+    // - found no way to entirely disable focus on list-view elements in single-item-selection mode without breaking list behavior
+    // - TODO GH#5400: the explicit implementation can be generalized by checking if the focused element is a descendant of palette
+    //         by iterating from it to the root of the Xaml tree using VisualTreeHelper::GetParent.
+    //         If this element is a descendant of palette register lost-focus-handler on it as well
+    // Arguments:
+    // - <unused>
+    // Return Value:
+    // - <none>
+    void CommandPalette::_lostFocusHandler(Windows::Foundation::IInspectable const& /* sender*/,
+                                           Windows::UI::Xaml::RoutedEventArgs const& /*args*/)
+    {
+        auto focusedElement = Input::FocusManager::GetFocusedElement(this->XamlRoot()).try_as<FrameworkElement>();
+        if (focusedElement && focusedElement != *this && focusedElement.Name() != L"_highlightedTextControl" && Visibility() != Visibility::Collapsed)
+        {
+            _dismissPalette();
+        }
```

Comment:
I've seen some strange things with the Focus event that make me feel like this should be easier.

If you add a LostFocus handler on _the root control_, and check the `sender` parameter instead of `GetFocusedElement`, you might find that you're getting LostFocus event fired for all contained controls. (I am not certain about this, but it _is_ a routed event!) That would make this easier: use `VisualTreeHelper` to walk the control's parent to see if it is us or is within us.

Reply:
@DHowett - Yeah.. I have a code with `VisualTreeHelper` iterating the parents. This is how I started, but putting the lost focus on the root of the palette was not sufficient. Are you suggesting to put in on `TerminalPage`?

Reply:
Also from the documentation of Lost Focus Event:

```
Specific Windows Runtime controls may have class-based handling for the LostFocus event. If so, the control probably has an override for the method OnLostFocus. Typically the event is marked handled by the class handler, and the LostFocus event is not raised for handling by any user code handlers on that control. Controls might handle the event in order to unload a visual state that displayed a focus rectangle in response to OnGotFocus.
```

Reply:
I can check if we can do something with  LosingFocus event that won't probably get swallowed the way LostFocus does